syntax = "proto3";
package proto;


message NodeConnector{
  map<int32,DataNode> input_data = 1;
}
message TaskNode{
  uint64 node_id = 1;
  NodeConnector node_structure = 2;
  FunctionParams function_param = 3;
}
message DataNode{
  uint64 node_id = 1;
  string uuid = 2;
  NodeConnector node_structure = 3;
}

message apmFile{
  repeated TaskNode nodes = 1;

}

//into raw data slot in NodeData
message FunctionParams{
  //General Main ServiceFunction
  string user_prompt = 1;
  // system prompt for LLM
  string system_prompt = 2;
}

